src/dev.rs

```diff
@@ -553,14 +553,18 @@ impl<F: Field + Group> Assignment<F> for MockProver<F> {
 
     fn fill_from_row(
         &mut self,
-        _: Column<Fixed>,
+        col: Column<Fixed>,
         from_row: usize,
-        _: Option<Assigned<F>>,
+        to: Option<Assigned<F>>,
     ) -> Result<(), Error> {
         if !self.usable_rows.contains(&from_row) {
             return Err(Error::not_enough_rows_available(self.k));
         }
 
+        for row in self.usable_rows.clone().skip(from_row) {
+            self.assign_fixed(|| "", col, row, || to.ok_or(Error::Synthesis))?;
+        }
```

Comment:
This commit results in the following test error:
```
thread 'dev::tests::bad_lookup' panicked at 'assertion failed: `(left == right)`
  left: `Err([Lookup { lookup_index: 0, location: InRegion { region: Region { index: 2, name: "Faulty synthesis" }, offset: 1 } }, Lookup { lookup_index: 0, location: OutsideRegion { row: 4 } }, Lookup { lookup_index: 0, location: OutsideRegion { row: 5 } }, Lookup { lookup_index: 0, location: OutsideRegion { row: 6 } }, Lookup { lookup_index: 0, location: OutsideRegion { row: 7 } }, Lookup { lookup_index: 0, location: OutsideRegion { row: 8 } }, Lookup { lookup_index: 0, location: OutsideRegion { row: 9 } }])`,
 right: `Err([Lookup { lookup_index: 0, location: InRegion { region: Region { index: 2, name: "Faulty synthesis" }, offset: 1 } }])`', src/dev.rs:1170:9
```

It's nice to see the improved error details from #431: we can clearly identify that the unintended errors are caused by lookup failures outside of any assigned region.